stdlib/Printf/src/Printf.jl

```diff
@@ -292,7 +292,13 @@ fmt(buf, pos, arg::AbstractFloat, spec::Spec{T}) where {T <: Ints} =
     bs = base(T)
     arg2 = toint(arg)
     n = i = ndigits(arg2, base=bs, pad=1)
-    x, neg = arg2 < 0 ? (-arg2, true) : (arg2, false)
+    one = oneunit(arg2)
+    neg = arg2 < 0
+    if typeof(arg2) <: Signed && !(typeof(arg2) <: BigInt)
+        x = neg ? unsigned(-(arg2+one))+unsigned(one) : unsigned(arg2)
```

Comment:
What about `x = unsigned(abs(arg2))`? That fixes the problematic case and avoids any conversion problems, since the `typemin` is handled by `unsigned` and all other negative values are handled by `abs` (since they're guaranteed to still fit into the original type).

Reply:
Thanks for the comment, I have updated to `x = unsigned(neg ? -arg2 : arg2)` to be closer to the original code with the same meaning.